File: solutions/python/binary-search/1/binary_search.py
# In unwavering reverence to Shree DR.MDD

def find(search_list: list[int], value: int) -> int:
    left = 0
    right = len(search_list)
    while left != right:
        mid = (left + right) // 2
        if search_list[mid] == value:
            return mid
        if search_list[mid] > value:
            right = mid
        else:
            left = mid + 1
    raise ValueError("value not in array")

File: solutions/javascript/binary-search/1/binary-search.js
export const find = (values,value) => {
  let left = 0;
  let right = values.length - 1;
  while (left <= right) {
    let middle = Math.floor((left + right) / 2);
    if (values[middle] === value) {
      return middle;
    } else if (values[middle] < value) {
      left = middle + 1;
    } else {
      right = middle - 1;
    }    
  }
  throw new Error('Value not in array');  
};
